### PR TITLE
Fix-component-undefined-svelte-v5

### PR DESCRIPTION
**Pull request by hermit99**
## Changes

Small code enhancement, which is easy to understand.

- Fix 'component is not defined' console error when unmounting svelte 5 component.

Console error: Screenshot 2024-10-01 110532
## Testing

Defensive code added and tested locally.

### CODE REVIEW

**Comment by martrapp** [comment]
Thanks @bluwy for pointing out!

@hermit99, thank you very much for providing this PR!
I'll have a look at #12116 and we might fix them both in the same way as the issues result from the same refactoring.

**Review by martrapp — CHANGES_REQUESTED** [review]
Hi @hermit99, the suggestion would be to revert the file and only move the addEventlistener from line 44 up to line 42 at the end of the else-block. That would ensure that `component` is always defined when referenced and would also fix the issue that the listener is currently added too often

File: .changeset/eighty-ligers-punch.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/svelte': patch
+---
+
+- fix: 'component is not defined' error when unmount svelte 5 component
```
Comment (martrapp) [inline .changeset/eighty-ligers-punch.md:5]:
```suggestion
Fixes an Reference Error that occurred during client transitions
```

**Review by martrapp — APPROVED** [review]

**Comment by hermit99** [comment]
Thanks @martrapp. Pardon me as I'm new to Astro's code base. Your suggestion makes more sense, and I really appreciate it ❤️

**Comment by martrapp** [comment]
Thank you very much for taking care! 💜 
The suggestion came from https://github.com/withastro/astro/pull/12116#issuecomment-2393768424

**Pull request merged**